server/gitrest/packages/gitrest-base/src/routes/git/blobs.ts

```diff
@@ -24,10 +28,14 @@ export function create(store: nconf.Provider, repoManagerFactory: IRepositoryMan
      * Retrieves the given blob from the repository
      */
     router.get("/repos/:owner/:repo/git/blobs/:sha", async (request, response, next) => {
-        const resultP = repoManagerFactory.open(getRepoManagerParamsFromRequest(request))
+        const repoManagerParams = getRepoManagerParamsFromRequest(request);
+        const resultP = repoManagerFactory.open(repoManagerParams)
             .then(async (repoManager) => repoManager.getBlob(
                 request.params.sha,
-            ));
+            )).catch((error) => {
+                logApiError(error, request, repoManagerParams);
+                throw error;
```

Comment:
Good idea! To centralize this conditional conversion to NetworkError, I added the check in `logApiError` and renamed it to `logAndThrowApiError`.

Reply:
Because some components in Fluid server expect some specific response codes (as we discussed, GetRef in GitManager expects 400) I made the default `NetworkError` have error code 400. Ideally, we would make sure each API in the routes/RepoManager implementations handles and explicitly uses those errors expected in the protocol. But since that is not yet the case, I used 400 to avoid disruptions. I made a comment in the code to remember about that later